### PR TITLE
error out by default beyond 10000 versions per object

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
error out by default beyond 10000 versions per object

## Motivation and Context
```
You've exceeded the limit on the number of versions you can create on this object
```

## How to test this PR?
Well, try to upload more than 10,000 versions per object.
```
mc: <ERROR> Failed to copy `/etc/hosts`. You've exceeded the limit on the number of versions you can create on this object
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [x] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/xl-storage-format-v2.go [hunk 1]
```diff
@@ -37,9 +38,30 @@ import (
...
+var maxObjectVersions = 10000
+
+func init() {
+	v := env.Get("_MINIO_OBJECT_MAX_VERSIONS", "")
+	if v != "" {
+		maxv, err := strconv.Atoi(v)
+		if err != nil {
+			logger.Info("invalid _MINIO_OBJECT_MAX_VERSIONS value: %s, defaulting to '10000'", v)
+			maxObjectVersions = 10000
+		}
```
Comment (vadmeste) [inline cmd/xl-storage-format-v2.go]: Better to do 'else' here (otherwise we can have two logger.Info()

**Review by vadmeste — COMMENTED** [review]
Minor thing, LGTM otherwise

**Review by klauspost — COMMENTED** [review]
I'd prefer to have this as a config value. Having to reboot the entire cluster to change the value is not very user friendly.

**Comment by harshavardhana** [comment]
> I'd prefer to have this as a config value. Having to reboot the entire cluster to change the value is not very user friendly.

It is kept the way it is right now since we don't know if this needs to be upgraded to a fully supported value. 

Since it is experimental at this point @klauspost - I really do not want users to keep changing this and play around with it at runtime for now until we bake this in properly.

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by shtripat — COMMENTED** [review]
Verified. LGTM

**Review by jiuker — APPROVED** [review]

**Pull request merged**